src/QsCompiler/QirGeneration/DebugInfoManager.cs

```diff
@@ -5,63 +5,94 @@
 using System.Collections.Generic;
 using System.Collections.Immutable;
 using System.IO;
+using System.Linq;
 using System.Reflection;
 using Microsoft.Quantum.QIR;
+using Microsoft.Quantum.QIR.Emission;
 using Microsoft.Quantum.QsCompiler.DataTypes;
+using Microsoft.Quantum.QsCompiler.SyntaxTokens;
 using Microsoft.Quantum.QsCompiler.SyntaxTree;
 using Ubiquity.NET.Llvm;
 using Ubiquity.NET.Llvm.DebugInfo;
+using Ubiquity.NET.Llvm.Instructions;
+using Ubiquity.NET.Llvm.Types;
+using Ubiquity.NET.Llvm.Values;
 
 namespace Microsoft.Quantum.QsCompiler.QIR
 {
+    using QsTypeKind = QsTypeKind<ResolvedType, UserDefinedType, QsTypeParameter, CallableInformation>;
+    using ResolvedTypeKind = QsTypeKind<ResolvedType, UserDefinedType, QsTypeParameter, CallableInformation>;
+
     /// <summary>
     /// This class holds shared utility routines for generating QIR debug information.
     /// It contains a reference to the GenerationContext that owns it because it needs
     /// access to the same shared state represented by the GenerationContext.
     /// </summary>
     internal sealed class DebugInfoManager
     {
-        /// <summary>
-        /// Dwarf version we are using for the debug info in the QIR generation
-        /// </summary>
-        private static readonly uint DwarfVersion = 4;
+// TODO: remove SECTIONS before merging to main
+// SECTION: Config values
+        private static class DebugInfoConfig
+        {
+            /// <summary>
+            /// Whether or not to emit debug information during QIR generation
+            /// </summary>
+            private static bool debugFlag = false;
```

Comment:
Since this is a static class, `DebugFlag` can be public and not readonly. That way you don't need the getter and setter either.

Reply:
If I make it public and not readonly I get the following error: [repo path]\src\QsCompiler\QirGeneration\DebugInfoManager.cs(40,32): warning SA1401: Field should be private [[repo path]\src\QsCompiler\QirGeneration\QirGeneration.csproj]. If I make it public and readonly the error goes away (but it shouldn't be readonly).

Reply:
In this case using the following syntax should get rid of the warning:
```csharp
public bool DebugSymbolsEnabled { get; set; } = false;
```

I changed the name to `DebugSymbolsEnabled` to make clear the intent of the property.

Reply:
Actually since this is a static class, the rest of the properties should follow the same pattern and not use `readonly`. Rather just having a `get` but not a `set`.